### PR TITLE
[SDRAN-901] Integration tests for topo

### PR DESCRIPTION
**Pull request by eroshiva**

### CODE REVIEW

File: test/utils/service.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	"github.com/onosproject/onos-lib-go/pkg/northbound"
+	"github.com/onosproject/onos-topo/pkg/store"
+	"google.golang.org/grpc"
+)
+
+var log = logging.GetLogger("northbound", "topo")
+
+// NewService returns a new topo Service
+func NewService(store store.Store) northbound.Service {
```
Comment (adibrastegarnia) [inline test/utils/service.go:31]: This is not needed. Please delete it. service is implemented in the NB of topo already.

File: test/utils/service.go [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
```
Comment (adibrastegarnia) [inline test/utils/service.go:17]: service.go should be removed completely. I don't understand why you need this file.

File: go.mod [hunk 3]
```diff
@@ -11,7 +11,8 @@ require (
 	github.com/onosproject/onos-api/go v0.7.66
 	github.com/onosproject/onos-lib-go v0.7.12
 	github.com/onosproject/onos-test v0.6.5
-	github.com/smartystreets/assertions v1.0.0 // indirect
+	github.com/prometheus/common v0.4.1
```
Comment (adibrastegarnia) [inline go.mod:14]: I don't know why this dep should show up here? Perhaps run go mod tidy.

File: test/topo/topo.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		ID: "1",
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, topoapi.ID("1"), gres.Object.ID)
+
+	log.Debugf("Listing all devices")
+	res, err := client.List(context.Background(), &topoapi.ListRequest{})
+	assert.NilError(t, err)
+	log.Debugf("Verifying that there are two devices stored")
+	assert.Equal(t, len(res.Objects) == 2 &&
```
Comment (adibrastegarnia) [inline test/topo/topo.go:67]: You can use assert.True to verfiy conditions.

File: test/topo/topo.go [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		(res.Objects[0].ID == "1" || res.Objects[1].ID == "1"), true)
+	//assert.Condition(t, func() bool {
```
Comment (adibrastegarnia) [inline test/topo/topo.go:69]: Please remove commented out code.

File: test/topo/topo.go [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+	//	return len(res.Objects) == 2 &&
+	//		(res.Objects[0].ID == "1" || res.Objects[1].ID == "1")
+	//})
+
+	log.Debugf("Updating first device")
+	obj := gres.Object
+	//obj.Attributes = make(map[string]string)
```
Comment (adibrastegarnia) [inline test/topo/topo.go:76]: You can use this utility functions associated with an object to Set/Get aspects and include them in tests: 
https://github.com/onosproject/onos-api/blob/master/go/onos/topo/topo.go

File: test/topo/topo.go [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+	"gotest.tools/assert"
+)
+
+var log = logging.GetLogger("topo")
+
+// TestAddRemoveDevice adds devices to the storage, lists and checks that they are in database and removes devices from the storage
+func (s *TestSuite) TestAddRemoveDevice(t *testing.T) {
+
+	log.Debugf("Creating connection")
```
Comment (adibrastegarnia) [inline test/topo/topo.go:32]: You should use t.Log for printing debug statement in test. Remove log from test.

File: test/topo/topo.go (hunk 4 above)
Comment (eroshiva) [reply test/topo/topo.go:67]: In the standart assertions package I use this function is not supported

File: go.mod (hunk 3 above)
Comment (eroshiva) [reply go.mod]: go mode tidy solved it :ok_hand:

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**